File: app/src/main/java/com/example/cdoherty/simplemvpspike/SignIn/PresenterSignIn.java
package com.example.cdoherty.simplemvpspike.SignIn;

/**
 * Created by cdoherty on 27/02/2018.
 */

public class PresenterSignIn implements SignInInteractor, SignInImplementation.SignInListener{

    private ViewSignIn viewSignIn;
    private ModelSignIn modelSignIn;

    public PresenterSignIn(ViewSignIn viewSignIn, ModelSignIn modelSignIn) {
        this.viewSignIn = viewSignIn;
        this.modelSignIn = modelSignIn;
    }

    @Override
    public void onSignInClick(String username, String password) {
        modelSignIn.signIn(username, password, this);

    }

    @Override
    public void onSignUpClick() {
        modelSignIn.signUp( this);
    }

    @Override
    public void onSignInReceived(boolean isSignedIn) {
        viewSignIn.signIn(isSignedIn);
    }

    @Override
    public void onSignUp() {
        viewSignIn.signUp();
    }

}

File: app/src/main/java/com/example/cdoherty/simplemvpspike/SignIn/Kotlin/PresenterSignIn.kt
package com.example.cdoherty.simplemvpspike.SignIn.Kotlin

import com.example.cdoherty.simplemvpspike.SignIn.ModelSignIn
import com.example.cdoherty.simplemvpspike.SignIn.SignInImplementation
import com.example.cdoherty.simplemvpspike.SignIn.SignInInteractor
import com.example.cdoherty.simplemvpspike.SignIn.ViewSignIn

/**
 * Created by cdoherty on 28/02/2018.
 */
class PresenterSignIn(private val viewSignIn: ViewSignIn, private val modelSignIn: ModelSignIn) : SignInInteractor, SignInImplementation.SignInListener {

    override fun onSignInClick(username: String, password: String) {
        modelSignIn.signIn(username, password, this)

    }

    override fun onSignUpClick() {
        modelSignIn.signUp(this)
    }

    override fun onSignInReceived(isSignedIn: Boolean) {
        viewSignIn.signIn(isSignedIn)
    }

    override fun onSignUp() {
        viewSignIn.signUp()
    }

}